[Excel] (Data types) Clarify data types learning path (#3741)
OfficeDev/office-js-docs-pr: docs/excel/excel-data-types-overview.md

## Resources

Explore data types with the following resources.

1. Learn data types basics in the [Excel data types core concepts](excel-data-types-concepts.md) article.
1. Install [Script Lab](../overview/explore-with-script-lab.md) in Excel and explore the **Data types** section in our **Samples** library.
1. Learn how to extend Excel beyond a 2-dimensional grid with entity data types in the [Use cards with entity value data types](excel-data-types-entity-card.md) article.
1. Try the [Create and explore data types in Excel](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/excel-data-types-explorer) sample in our [OfficeDev/Office-Add-in-samples](https://github.com/OfficeDev/Office-Add-in-samples) repository.

> > [!NOTE]

Reviewer: This note is a repeat of bullet 2.

Author: I'm hesitant to remove the note because I originally added it in response to feedback (uncertainty about how to get started developing data types). I think the point of the note in this location is to emphasize that the reader can skip step 1 and move straight to Script Lab, if they prefer a sandbox type learning style. I agree that it's somewhat redundant, though. What do you think?

Reviewer: What if the note is moved right after the first paragraph ("skip the reading and try it out") or after the picture (so it'll be visually inline with the image)?

Author: Done! Moved it to after the image. Thanks for all of the suggestions!

> 

Reviewer: Consider adding a screenshot to this article to help show what data types are. It's one of those features that makes people say "Wow, I didn't know Excel could do that."

Data types expand Excel JavaScript API support beyond the four original data types (string, number, boolean, and error) to include [web images](excel-data-types-concepts.md#web-image-values), [formatted number values](excel-data-types-concepts.md#formatted-number-values), [entities](excel-data-types-concepts.md#entity-values), arrays within entities, and improved [error data types](excel-data-types-concepts.md#improved-error-support) as flexible data structures. These types, which power many [linked data types](https://support.microsoft.com/office/what-linked-data-types-are-available-in-excel-6510ab58-52f6-4368-ba0f-6a76c0190772) experiences, allow for precision and simplicity during add-in calculations and extend the potential of Excel add-ins beyond a 2-dimensional grid.

> Take the following steps to complete the data types learning path.

Reviewer: "Learning path" sounds like an industry term. Consider something like this:
```suggestion
Learn all about data types with these resources.
```

> 1. Learn data types API basics in the [Excel data types core concepts](excel-data-types-concepts.md) article.

Reviewer: Consider removing "API" to make this more conversational.  
```suggestion
1. Learn data types basics in the [Excel data types core concepts](excel-data-types-concepts.md) article.
```

1. Install [Script Lab](../overview/explore-with-script-lab.md) in Excel and explore the **Data types** section in our **Samples** library.

> 1. Learn how entity data types extend Excel add-ins beyond a 2-dimensional grid in the [Use cards with entity value data types](excel-data-types-entity-card.md) article.

Reviewer: Two suggestions:
1. Lead with the pitch, rather than the feature.
2. Since add-ins already expand Excel beyond the grid, consider couching this in terms of Excel itself.
```suggestion
1. Learn how to extend Excel beyond a 2-dimensional grid with entity data types in the [Use cards with entity value data types](excel-data-types-entity-card.md) article.
```

> 1. Try out the [Create and explore data types in Excel](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/excel-data-types-explorer) sample in our [OfficeDev/Office-Add-in-samples](https://github.com/OfficeDev/Office-Add-in-samples) repository.

Reviewer: ```suggestion
1. Try the [Create and explore data types in Excel](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/excel-data-types-explorer) sample in our [OfficeDev/Office-Add-in-samples](https://github.com/OfficeDev/Office-Add-in-samples) repository.
```

> [!NOTE]

> > To start experimenting with data types right away, install [Script Lab](../overview/explore-with-script-lab.md) in Excel and check out the **Data types** section in our **Samples** library. You can also explore the Script Lab samples in our [OfficeDev/office-js-snippets](https://github.com/OfficeDev/office-js-snippets/tree/prod/samples/excel/20-data-types) repository.

Reviewer: The See also section now feels redundant. Is there value in removing it, and instead, giving you previous list a header like "Resources"?
